packages/fury-adapter-remote/lib/adapter.js

```diff
@@ -46,18 +51,28 @@ class FuryRemoteAdapter {
   constructor(options) {
     this.name = name;
     this.options = options || defaultOptions;
-    const parseMediaTypes = this.options.parseMediaTypes || [];
-    const serializeMediaTypes = this.options.serializeMediaTypes || [];
-
-    this.mediaTypes = parseMediaTypes.concat(serializeMediaTypes);
+    this.mediaTypes = this.options.mediaTypes || [];
   }
 
-  detect(source) {
-    return this.mediaTypes.includes(deckardcain.identify(source));
+  detect(source, method) {
+    const mediaType = deckardcain.identify(source);
+    if (Array.isArray(this.mediaTypes) && this.mediaTypes.includes(mediaType) && (method === undefined || this[method])) {
+      return true;
+    } if (typeof this.mediaTypes === 'object') {
+      if (method !== undefined) {
+        return (this.mediaTypes[method] && this.mediaTypes[method].includes(mediaType) && this[method]);
+      }
+      const mediaTypes = [];
+      Object.keys(this.mediaTypes).forEach((key) => {
+        mediaTypes.concat(this.mediaTypes[key]); // the value of the current key.
+      });
+      return mediaTypes.includes(mediaType);
+    }
+    return false;
   }
 
-  parse({ source, minim }, cb) {
-    const inputMediaType = detectMediaType(source, this.options.defaultInputMediaType);
+  parse({ source, minim, mediaType }, cb) {
+    const inputMediaType = mediaType || detectMediaType(source, this.options.defaultInputMediaType);
```

Comment:
`defaultInputMediatype` was removed in the rest of PR. Should it still be here?

Reply:
You are right, I miss it